packages/server/src/api/controllers/row/ExternalRequest.ts

```diff
@@ -743,9 +745,19 @@ export class ExternalRequest<T extends Operation> {
 
     // aliasing can be disabled fully if desired
     const aliasing = new sdk.rows.AliasTables(Object.keys(this.tables))
-    let response = env.SQL_ALIASING_DISABLE
-      ? await getDatasourceAndQuery(json)
-      : await aliasing.queryWithAliasing(json, makeExternalQuery)
+    let response: DatasourcePlusQueryResponse
+    // there's a chance after input processing nothing needs updated, so pass over the call
+    // we might still need to perform other operations like updating the foreign keys on other rows
+    if (
+      this.operation === Operation.UPDATE &&
+      Object.keys(row || {}).length === 0
+    ) {
+      response = [config.row!]
+    } else {
```

Comment:
I don't think I fully understand this 🤔 So if the row is empty we return `[{}]`?

Reply:
Ah - in this case the `row` is the processed row, which may have been stripped down if nothing has changed, while `config.row` is unprocessed - I'll rename to make it more obvious.